src/object-pool.ts

```diff
@@ -1,10 +1,287 @@
-import {ArmorObjectPoolOptions} from './object-pool-options';
-import {ArmorObjectPoolState} from './object-pool-state';
+import ArmorCollection from './collection';
+import ArmorCollectionSelector from './selector';
+import ArmorObjectPoolInstance from './object-pool-instance';
+import ArmorObjectPoolOptions from './object-pool-options';
+import ArmorObjectPoolState from './object-pool-state';
 
-export class ArmorObjectPool<T> {
-	public readonly state: ArmorObjectPoolState;
+export default class ArmorObjectPool<T> implements ArmorCollection<T> {
+	public state: ArmorObjectPoolState<T>;
+	public startSize: number = 10;
+	public readonly poolObj: ArmorObjectPoolInstance<T>;
 
-	constructor(options?: ArmorObjectPoolOptions) {
-		this.state = new ArmorObjectPoolState(options);
+	constructor(poolObj: ArmorObjectPoolInstance<T>, options?: ArmorObjectPoolOptions<T>) {
+		if (typeof poolObj !== 'function') {
+			throw new Error('Must have a comparator function for priority queue to operate properly');
+		}
+		this.poolObj = poolObj;
+
+		this.state = {
+			type: 'opState',
+			elements: [],
+			objectCount: 0,
+			maxSize: 1000,
+			autoIncrease: false,
+			increaseFactor: 2,
+			increaseBreakPoint: 0.8
+		};
+
+		if (options) {
+			this.parseOptions(options);
+		}
+
+		this.increase(this.startSize);
+	}
+	public parseOptions(options: ArmorObjectPoolOptions<T>): void {
+		if (!options) {
+			return;
+		}
+
+		if (options.state !== undefined) {
+			this.parseOptionsState(options.state);
+		}
+
+		if (options.startSize !== undefined) {
+			this.parseOptionsStartSize(options.startSize);
+		}
+	}
+	public parseOptionsStartSize(startSize: number): void {
+		if (!this.isInteger(startSize)) {
+			return;
+		}
+		if (startSize < 0) {
+			return;
+		}
+
+		this.startSize = startSize;
+	}
+	public parseOptionsState(state: ArmorObjectPoolState<T> | string): void {
+		if (!state) {
```

Comment:
If state is undefined for some reason, what values are the state properties initialized to?